torchtext/data/iterator.py

```diff
@@ -65,7 +65,8 @@ def __init__(self, dataset, batch_size, sort_key=None, device=None,
             logger.warning("The `device` argument should be set by using `torch.device`"
                            + " or passing a string as an argument. This behavior will be"
                            + " deprecated soon and currently defaults to cpu.")
-            device = torch.device('cpu')
```

Comment:
This is not correct. The `device` is under `if isinstance` statement. For `None` device, it's no way to enter it.